editoast/src/schema/coreclient/pathfinding.rs

```diff
@@ -0,0 +1,180 @@
+use derivative::Derivative;
+use geos::geojson::{Geometry, Value::LineString};
+use serde::{Deserialize, Serialize};
+
+use crate::schema::{
+    self,
+    rolling_stock::{EffortCurves, Gamma, RollingResistance, RollingStockMetadata},
+    Direction,
+};
+use schema::rolling_stock::RollingStock as RollingStockSchema;
+
+use super::AsCoreRequest;
+
+pub type PathfindingWaypoints = Vec<Vec<Waypoint>>;
+
+#[derive(Debug, Serialize)]
+pub struct PathfindingRequest {
+    infra: i64,
+    expected_version: String,
+    waypoints: PathfindingWaypoints,
+    rolling_stocks: Vec<RollingStock>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct Waypoint {
+    track_section: String,
+    direction: Direction,
+    offset: f64,
+}
+
+#[derive(Debug, Serialize)]
+pub struct RollingStock {
+    name: String,
+    version: String,
+    effort_curves: EffortCurves,
+    base_power_class: String,
+    length: f64,
+    max_speed: f64,
+    startup_time: f64,
+    startup_acceleration: f64,
+    comfort_acceleration: f64,
+    gamma: diesel_json::Json<Gamma>,
+    inertia_coefficient: f64,
+    features: Vec<String>,
+    mass: f64,
+    rolling_resistance: diesel_json::Json<RollingResistance>,
+    loading_gauge: String,
+    metadata: diesel_json::Json<RollingStockMetadata>,
+    power_restrictions: Option<serde_json::Value>,
+}
+
+#[derive(Debug, Clone, Derivative, Deserialize, Serialize)]
+#[derivative(Default)]
+pub struct PathfindingResponse {
+    #[derivative(Default(value = "Geometry::new(LineString(Default::default()))"))]
+    pub geographic: Geometry,
+    #[derivative(Default(value = "Geometry::new(LineString(Default::default()))"))]
+    pub schematic: Geometry,
+    pub route_paths: Vec<RoutePath>,
+    pub path_waypoints: Vec<PathWaypoint>,
+    pub warnings: Vec<Warning>,
+}
+
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct RoutePath {
+    pub route: String,
+    pub signaling_type: String,
+    pub track_sections: Vec<RouteTrackSection>,
+}
+
+#[derive(Debug, Clone, Derivative, Deserialize, Serialize)]
+#[derivative(Default)]
+pub struct RouteTrackSection {
+    pub track: String,
+    #[derivative(Default(value = "Direction::StartToStop"))]
+    pub direction: Direction,
+    pub begin: f64,
+    pub end: f64,
+}
```

Comment:
Isn't that a `DirectionalTrackRange` ? https://github.com/DGEXSolutions/osrd/blob/5cfdb304d7e5ab9d8d2815ba645d1447af26fc56/editoast/src/schema/mod.rs#L208-L219